spec.md

```diff
@@ -1448,6 +1451,51 @@ The CO MUST implement the specified error recovery behavior when it encounters t
 |-----------|-----------|-------------|-------------------|
 | Invalid `starting_token` | 10 ABORTED | Indicates that `starting_token` is not valid. | Caller SHOULD start the `ListVolumes` operation again with an empty `starting_token`. |
 
+#### `GetVolume`
+
+A Controller Plugin MUST implement this `GetVolume` RPC call if it has `GET_VOLUME` capability.
+
+A Controller Plugin MUST support the `volume_health` field in the `status` field in `GetVolumeResponse` if it has `GET_VOLUME_HEALTH` capability.
+
+`GetVolumeResponse` should contain current information of a volume if it exists. If the volume does not exist any more, `GetVolume` should return gRPC error code `NOT_FOUND`.
+
+The `VolumeHealth` message in `VolumeStatus` field in `GetVolumeResponse` can indicate additional health information for the volume.
+
+Whether the volume is still attached to a node is already handled by `LIST_VOLUMES_PUBLISHED_NODES` controller capability. This capability MAY be used by CO to reconcile - re-attach the volume if the volume is not attached any more.
+
+```protobuf
+message GetVolumeRequest {
+  // Identity information for a specific volume. This field is
+  // REQUIRED. GetVolume will return with current volume information.
+  string volume_id = 1;
+}
+
+message GetVolumeResponse {
+  message VolumeStatus{
+    // volume_health shows error conditions reported by the SP.
+    // This field MUST be specified if the
+    // GET_VOLUME_HEALTH controller capability is supported.
+    repeated VolumeHealth volume_health = 1;
+  }
+
+  // This field is REQUIRED
+  Volume volume = 1;
+
+  // This field is OPTIONAL. This field MUST be specified if the
+  // GET_VOLUME_HEALTH controller capability is supported.
+  VolumeStatus status = 2;
+}
+```
+
+##### GetVolume Errors
+
+If the plugin is unable to complete the GetVolume call successfully, it MUST return a non-ok gRPC code in the gRPC status.
+
+##### VolumeHealth from Controller vs Node Plugins
+
+If a SP provides `VolumeHealth` information in both Controller and Node Plugins, information provided SHALL not be duplicate. `VolumeHealth` information from the Controller Plugin SHALL be from the storage system but not specific for one node, while information from the Node Plugin SHALL be specific for that node.
```

Comment:
"SHALL not be duplicate" -- what if it is? is the CO expected to fail the RPC call? this sounds arbitrarily restrictive, though I think I understand the goal of this wording. please re-phrase.

nit: one sentence per line please

Reply:
Sure